src/etl/sds/trigger/bulk/steps.py

```diff
@@ -42,49 +39,56 @@ def _validate_database_is_empty(data, cache: Cache):
 
 def _create_state_machine_input(data, cache):
     _, source_key = data[StepChain.INIT]
-    return StateMachineInput.bulk(changelog_number=Path(source_key).stem)
+    state_machine_input = StateMachineInput.bulk(changelog_number=Path(source_key).stem)
+    return state_machine_input
 
 
-def _copy_to_state_machine(data, cache: Cache):
-    source_bucket, source_key = data[StepChain.INIT]
-    return cache["s3_client"].copy_object(
-        Bucket=source_bucket,
-        Key=WorkerKey.EXTRACT,
-        CopySource=f"{source_bucket}/{source_key}",
-    )
-
-
-def _copy_to_history(data, cache: Cache):
+def _copy_to_intermediate_history_file(data, cache: Cache):
     source_bucket, source_key = data[StepChain.INIT]
     state_machine_input: StateMachineInput = data[_create_state_machine_input]
     return cache["s3_client"].copy_object(
         Bucket=source_bucket,
-        Key=f"history/{state_machine_input.etl_type}/{source_key}",
+        Key=f"history/{state_machine_input.name}",
         CopySource=f"{source_bucket}/{source_key}",
     )
 
 
-def _delete_object(data, cache: Cache):
+def _delete_trigger_object(data, cache: Cache):
     source_bucket, source_key = data[StepChain.INIT]
     return cache["s3_client"].delete_object(Bucket=source_bucket, Key=source_key)
 
 
-def _start_execution(data, cache):
+def _publish_message_to_sqs_queue(data, cache: Cache):
+    sqs_client = boto3.client("sqs")
+
+    # Convert the state_machine_input to JSON
     state_machine_input: StateMachineInput = data[_create_state_machine_input]
-    return start_execution(
-        step_functions_client=cache["step_functions_client"],
-        state_machine_arn=cache["state_machine_arn"],
-        state_machine_input=state_machine_input,
+    state_machine_name = state_machine_input.name
+    message_body = {
+        "changelog_number_start": state_machine_input.changelog_number_start,
+        "changelog_number_end": state_machine_input.changelog_number_end,
+        "etl_type": state_machine_input.etl_type.value,
+        "timestamp": state_machine_input.timestamp,
+        "name": state_machine_name,
+    }
+    message_body_json = json.dumps(message_body)
```

Comment:
is this any different to

```python
state_machine_input.json()
```
?

Reply:
Ah yeah that is much nicer! It just doesn't include the name parameter so I've added a little function to the StateMachine class to return the json including the name as well.